api/v1alpha1/project_config_types.go

```diff
@@ -88,16 +88,19 @@ type PromotionPolicy struct {
 
 // WebhookReceiverConfig describes the configuration for a single webhook
 // receiver.
+//
+// +kubebuilder:validation:XValidation:message="WebhookReceiverConfig must have exactly one of github or gitlab set",rule="has(self.github) ? !has(self.gitlab) : has(self.gitlab)"
```

Comment:
@fuskovic I sent you a message offline. I can see more clearly now why you were reluctant about this line. It works great for two fields, but the expression becomes extremely cumbersome for three or more.

Since it's working for now, we can merge this and we'll just have to fall back on using a validating webhook for this when one of the other in-flight PRs adds a third field.

Reply:
You can do the honors.